lib/vintage_net_wizard/wifi_configuration.ex

```diff
@@ -102,6 +102,9 @@ defmodule VintageNetWizard.WiFiConfiguration do
   defp maybe_put_priority(vn_config, nil), do: vn_config
   defp maybe_put_priority(vn_config, priority), do: Map.put(vn_config, :priority, priority)
 
+  defp maybe_put_psk(vn_config, nil), do: vn_config
```

Comment:
Do we care if the password is an empty string `""`? Would that be considered _not_ as password and should be excluded?

Reply:
I am really not sure what WPA supplicant would do if we tried to send a password (even if empty) to connect to a no password SSID. However, I do know if we don't send a password to a non-secured SSID via WPA supplicant, it would work.

Reply:
There are some restrictions to WiFi passwords. `vintage_net` will raise on the config if the following aren't met:

1. The passphrase needs to be all ASCII characters (values between 32 and 126, inclusive)
2. PSKs are exactly 64 characters and much be hex digits. People don't normally do this, but `vintage_net` and the `wpa_supplicant` allow it.
3. Passphrases are between 8 and 63 characters. `vintage_net` raises if the password is > 63 characters and not a PSK, but doesn't raise on less than 8 characters. It probably should since that's a requirement of IEEE 802.11i, but I could have sworn that I've used short passwords and been ok...

This function could be used to validate a passphrase: https://hexdocs.pm/vintage_net/0.6.0/VintageNet.WiFi.WPA2.html#to_psk/2. The `ssid` part could be set to something bogus. A nicer option would be to expose the passphrase validation.

Thinking about it now, `vintage_net` really should raise on short passphrases. 8 is the answer in the spec. Empty is definitely wrong.

Reply:
This [PR](https://github.com/nerves-networking/vintage_net/pull/134) adds `valdiate_passphrase` to `vintage_net` once released I can swing back through and update this area to use it.

Reply:
:shipit: